catroidTest/src/org/catrobat/catroid/uitest/ui/activity/PocketMusicTest.java

```diff
@@ -86,6 +89,40 @@ public void testNoteGridElement() {
 				.musicdroid_note_grid));
 	}
 
+	public void testAllButtonToggle() {
+		prepareTest();
+		solo.waitForActivity(PocketMusicActivity.class.getSimpleName());
+
+		TrackView trackView = (TrackView) solo.getCurrentActivity().findViewById(R.id.musicdroid_note_grid);
+		for (TrackRowView trackRowView : trackView.getTrackRowViews()) {
+			for (NoteView noteView : trackRowView.getNoteViews()) {
+				solo.clickOnView(noteView);
+				solo.sleep(200);
+				assertTrue("Button not toggled", noteView.isToggled());
+			}
+		}
+
+		for (TrackRowView trackRowView : trackView.getTrackRowViews()) {
+			for (NoteView noteView : trackRowView.getNoteViews()) {
+				solo.clickOnView(noteView);
+				solo.sleep(200);
+				assertTrue("Button not toggled", !noteView.isToggled());
+			}
+		}
+	}
+
+	public void testButtonCount() {
+		prepareTest();
+		solo.waitForActivity(PocketMusicActivity.class.getSimpleName());
+
+		TrackView trackView = (TrackView) solo.getCurrentActivity().findViewById(R.id.musicdroid_note_grid);
+		assertEquals("TrackView size invalid", trackView.getTrackRowViews().size(), TrackView.ROW_COUNT);
+
+		for (TrackRowView trackRowView : trackView.getTrackRowViews()) {
+			assertEquals("TrackRowView size invalid", trackRowView.getNoteViews().size(), trackRowView.getTactCount());
```

Comment:
just a small thing: the `assertEquals()` parameter format should look like this: `assertEquals(text, actualValue, expectedValue)`. You could switch your parameters around to be apply this as well. This makes debugging failed tests easier